### PR TITLE
add test trainee (create method)

### PR DESCRIPTION
**Pull request by AlKaliada**

### CODE REVIEW

File: src/test/java/com/exadel/project/trainee/controller/TraineeCrudTest.java [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+            Trainee trainee = traineeTestData.getTestTrainee();
+            Trainee savedTrainee = traineeTestData.getTestTrainee();
+            savedTrainee.setId(null);
+            InterviewPeriod interviewPeriod = traineeTestData.getTestInterviewPeriod();
+            InterviewPeriod savedInterviewPeriod = traineeTestData.getTestInterviewPeriod();
+            savedInterviewPeriod.setId(null);
+            AdditionalInfo additionalInfo = traineeTestData.getTestAdditionalInfo();
+            Internship internship = traineeTestData.getTestInternship();
+
+            doReturn(null).when(traineeRepository).findTraineeByEmail(anyString());
```
Comment (ValentinKhorunzhyn) [inline src/test/java/com/exadel/project/trainee/controller/TraineeCrudTest.java:84]: When we want to create an trainee, we must check if we have created an trainee with the same email. can we pass traineeDTO.getEmail to the find method?

File: src/test/java/com/exadel/project/trainee/controller/TraineeCrudTest.java [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+            doReturn(internship).when(internshipService).getEntityById(anyLong());
```
Comment (ValentinKhorunzhyn) [inline src/test/java/com/exadel/project/trainee/controller/TraineeCrudTest.java:85]: The person is trying to register for internship number 1. Maybe we should pass to the method - .getEntityById(1L);

File: src/test/java/com/exadel/project/trainee/controller/TraineeCrudTest.java (hunk 1 above)
Comment (AlKaliada) [reply src/test/java/com/exadel/project/trainee/controller/TraineeCrudTest.java]: thanks, fixed

File: src/test/java/com/exadel/project/trainee/controller/TraineeCrudTest.java (hunk 2 above)
Comment (AlKaliada) [reply src/test/java/com/exadel/project/trainee/controller/TraineeCrudTest.java]: fixed

File: src/test/java/com/exadel/project/trainee/controller/TraineeCrudTest.java [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Autowired
+    private TraineeTestData traineeTestData;
+
+    @Nested
+    @DisplayName("Check Trainee's create method")
+    class testCreateTraineeMethod{
```
Comment (serkushner) [inline src/test/java/com/exadel/project/trainee/controller/TraineeCrudTest.java:65]: Not a method

Comment (AlKaliada) [reply src/test/java/com/exadel/project/trainee/controller/TraineeCrudTest.java]: thanks, fixed

**Review by serkushner — APPROVED** [review]

**Review by ValentinKhorunzhyn — APPROVED** [review]

**Pull request merged**